src/TextField/TextField.js

```diff
@@ -105,7 +105,7 @@ const getStyles = (props, context, state) => {
  * @returns True if the string provided is valid, false otherwise.
  */
 function isValid(value) {
-  return value !== '' && value !== undefined && value !== null;
+  return value !== '' && value !== undefined && value !== null && !(Array.isArray(value) && value.length === 0);
```

Comment:
Yes. I agree that it look a little odd, but I added it so SelectField's value can be initialized to an empty array (as well as null) which feels more natural when multiple is true. Without this test TextField always thinks it has a value and hintText and floatingLabelText doesn't behave as expected.